stl/inc/future

```diff
@@ -282,7 +300,23 @@ public:
             _Rethrow_future_exception(_Exception);
         }
 
-        return _Result;
+        if constexpr (is_default_constructible_v<_Ty>) {
+            return _Result;
+        } else {
+            return _Result._Held_value;
+        }
+    }
+
+    template <class _Ty2>
+    void _Emplace_result(_Ty2&& _Val) {
+        // TRANSITION: _Associated_state ctor will move or copy assign if _Result was default constructed
+        // rather than move or copy construct

```

Comment:
Ditto present tense, and it's also confusing that this is talking about the `_Associated_state` constructor which has already completed. I _think_ we want:
```suggestion
        // TRANSITION, incorrectly assigns _Result when _Ty is default constructible
```